File: src/api/envVar.ts
import { invoke } from '@tauri-apps/api/core';
import type {
  EnvironmentVariable,
  CreateEnvVarRequest,
  UpdateEnvVarRequest,
  EnvVarCategory,
} from '../types';

export const envVarApi = {
  async getEnvironmentVariables(): Promise<EnvironmentVariable[]> {
    return invoke('get_env_vars');
  },

  async getEnvVarCategories(): Promise<EnvVarCategory[]> {
    return invoke('get_env_var_categories');
  },

  async createEnvVar(req: CreateEnvVarRequest): Promise<EnvironmentVariable> {
    return invoke('create_env_var', { req });
  },

  async updateEnvVar(id: string, req: UpdateEnvVarRequest): Promise<EnvironmentVariable> {
    return invoke('update_env_var', { id, req });
  },

  async deleteEnvVar(id: string): Promise<void> {
    return invoke('delete_env_var', { id });
  },

  async deleteEnvVars(ids: string[]): Promise<void> {
    return invoke('delete_env_vars', { ids });
  },

  async syncEnvVar(id: string): Promise<EnvironmentVariable> {
    return invoke('sync_env_var', { id });
  },

  async syncAllEnvVars(): Promise<EnvironmentVariable[]> {
    return invoke('sync_all_env_vars');
  },

  async validateEnvVarName(name: string): Promise<{ valid: boolean; error?: string }> {
    return invoke('validate_env_var_name', { name });
  },

  async openEnvVarSettings(): Promise<void> {
    return invoke('open_env_var_settings');
  },
};

File: src-tauri/src/commands/env_var.rs
use crate::error::Result;
use crate::services::env_var::{
    CreateEnvVarRequest, EnvVarCategory, EnvironmentVariable, UpdateEnvVarRequest,
    ValidateNameResult,
};
use crate::services::EnvVarService;

#[tauri::command]
pub fn get_env_vars() -> Result<Vec<EnvironmentVariable>> {
    EnvVarService::get_env_vars()
}

#[tauri::command]
pub fn get_env_var_categories() -> Result<Vec<EnvVarCategory>> {
    EnvVarService::get_env_var_categories()
}

#[tauri::command]
pub fn create_env_var(req: CreateEnvVarRequest) -> Result<EnvironmentVariable> {
    EnvVarService::create_env_var(req)
}

#[tauri::command]
pub fn update_env_var(id: String, req: UpdateEnvVarRequest) -> Result<EnvironmentVariable> {
    EnvVarService::update_env_var(id, req)
}

#[tauri::command]
pub fn delete_env_var(id: String) -> Result<()> {
    EnvVarService::delete_env_var(id)
}

#[tauri::command]
pub fn delete_env_vars(ids: Vec<String>) -> Result<()> {
    EnvVarService::delete_env_vars(ids)
}

#[tauri::command]
pub fn sync_env_var(id: String) -> Result<EnvironmentVariable> {
    EnvVarService::sync_env_var(id)
}

#[tauri::command]
pub fn sync_all_env_vars() -> Result<Vec<EnvironmentVariable>> {
    EnvVarService::sync_all_env_vars()
}

#[tauri::command]
pub fn validate_env_var_name(name: String) -> Result<ValidateNameResult> {
    Ok(EnvVarService::validate_env_var_name(&name))
}

#[tauri::command]
pub fn open_env_var_settings() -> Result<()> {
    EnvVarService::open_env_var_settings()
}
